### PR TITLE
Update Python API for archetype development

### PR DESCRIPTION
**Pull request by abachma2**
This PR reflects changes made in [Cyclus PR #1594](https://github.com/cyclus/cyclus/pull/1594) to add some functionality to the Python API for creating material requests. There are also some small typo fixes on a few other pages around the site. 

This should *not* be merged until after Cyclus PR #1594 is merged.

### CODE REVIEW

File: source/arche/dre.rst [hunk 1]
```diff
@@ -96,11 +116,11 @@ be:
         recipe_b = self.context().get_recipe("recipeB")
         target_b = ts.Material.create_untracked(request_qty, recipe_b)
         # commodity mapping to request target
-        commods = {"FuelA": target_a, "FuelB": target_b}
+        commods = {"FuelA": target_a}, {"FuelB": target_b}
```
Comment (gonuke) [inline source/arche/dre.rst:119]: should this be a list?
```suggestion
        commods = [{"FuelA": target_a}, {"FuelB": target_b}]
```

File: source/arche/dre.rst [hunk 2]
```diff
@@ -114,10 +134,27 @@ be:
...
         return port
 
+        # In this example, both the FuelA and FuelB commodities can meet the 
+        # material request, but FuelA is preferred. So we define a preference 
+        # for each commodity. The larger the preference value, the 
+        # more the commodity is preferred. Placing the dictionaries in
+        # a list makes them mutual requests:
+        commods = [{"FuelA": target_a, "preference":2}, 
+                   {"FuelB": target_b, "preference":1}]
+        port = {"commodities":commods, "constraints":request_qty
```
Comment (gonuke) [inline source/arche/dre.rst:144]:
```suggestion
        port = {"commodities":commods, "constraints":request_qty}
```

File: source/arche/dre.rst [hunk 3]
```diff
@@ -114,10 +134,27 @@ be:
...
+
+        # If you want the requests to be exclusive, then you have to indicate 
+        # that:
+        commods = [{"FuelA": target_a, "preference":2, "exclusive":True}, 
+                   {"FuelB": target_b, "preference":1, "exclusive":True}]
+        port = {"commodities":commods, "constraints":request_qty
```
Comment (gonuke) [inline source/arche/dre.rst:150]:
```suggestion
        port = {"commodities":commods, "constraints":request_qty}
```

File: source/arche/tutorial_py/toolkit.rst [hunk 4]
```diff
@@ -51,7 +51,7 @@ This creates a state variable named ``inventory`` that is based on the
 ``ResBuf`` class.  We'll explain how buffers
 work in just a moment.  A ResBuf object and its ``ts.Inventory`` state var has special
 handling by the Python ``Agent`` class from which ``Facility`` inherits. It will not appear
-in the schema and therefore will not appear in the Cycic UI either.
+in the schema and therefore will not appear in the Cyclus UI either.
```
Comment (gonuke) [inline source/arche/tutorial_py/toolkit.rst:54]: I'm not sure how to correct this.  It was referring to a now-deprecated UI called "Cycic".  While we may not want to leave references to that lying around, I'm not sure that this is correct, either.

**Review by gonuke — CHANGES_REQUESTED** [review]
A few little things... does this mean that you are actually able to build the website still?

**Comment by abachma2** [comment]
I can at least build the preview locally with `make gh-preview`. I don't know about a full build that would be live on the webpages.

File: source/arche/dre.rst (hunk 1 above)
Comment (abachma2) [reply source/arche/dre.rst:119]: No. If you don't include "commodities" as one of the keys, then it expects the value to be a Material or Resource. I intentionally left this structure as the most basic thing you can return from this function in the python API.

File: source/arche/tutorial_py/toolkit.rst (hunk 4 above)
Comment (abachma2) [reply source/arche/tutorial_py/toolkit.rst:54]: We can just take out that clause then. I don't think I've seen any other references to this.

File: source/arche/dre.rst (hunk 1 above)
Comment (gonuke) [reply source/arche/dre.rst:119]: Then I'm not sure what the comma means here?  They aren't grouped into a data structure in any way

Comment (abachma2) [reply source/arche/dre.rst:119]: You are correct. It should be all one dictionary. Thanks for catching that. I've corrected it.

**Review by gonuke — APPROVED** [review]

**Comment by gonuke** [comment]
Thanks @abachma2

**Comment by gonuke** [comment]
Now we have to figure out how to publish an update successfully... :)

**Comment by abachma2** [comment]
I'll work on that next week and ping y'all in the slack if I run into big issues.

**Pull request merged**